### PR TITLE
Issue #125 - Replacing labels with uri-safe-values where possible

### PR DESCRIPTION
**Pull request by robons**
Upon serialising a QbCube to CSV-W, we now replace all of the known labels with their appropriate local URI-safe values. This ensures that when a user defines a `NewQbDimension.from_data(...)`, we map their data to the new URI-safe values which define the codes. Similar mappings are written for `NewQbAttributeValue`s, as well as `NewQbUnit`s inside units columns and `NewQbMeasure`s inside measure dimension columns.

Satisfies issue #125

### CODE REVIEW

File: csvqb/csvqb/tests/unit/utils/qb/test_standardise.py [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+        "Some Dimension": {"a", "b", "c"},
+        "Some Resource Attribute": {"r-a", "r-b", "r-c"},
+        "Measure": {"meas-a", "meas-b", "meas-c"},
+        "Unit": {"unit-a", "unit-b", "unit-c"},
+    }
+    expected_non_categorical_columns = {"Observed Value", "Some Literal Attribute"}
+
+    for (column_name, expected_categories) in map_col_to_expected_categories.items():
+        values = cube.data[column_name].values
+        assert isinstance(values, Categorical)
```
Comment (santhosh-thangavel) [inline csvqb/csvqb/tests/unit/utils/qb/test_standardise.py:65]: Is there a chance for the values to be an empty string or Null value? If so what would be the outcome?

Comment (robons) [reply csvqb/csvqb/tests/unit/utils/qb/test_standardise.py:65]: It shouldn't happen, but if it ever did, it'd cause the unit-test to fail which I think is perfectly fine - we'll know something is wrong.

File: csvqb/csvqb/tests/unit/utils/qb/test_standardise.py [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+    ensure_qbcube_data_is_categorical,
+    convert_data_values_to_uri_safe_values,
+)
+
+
+def test_qbcube_data_is_categorical():
+    """
+    Ensure that when we convert a QbCube's dataframe columns to categorical, it is done for the appropriate columns.
+    """
+    data = pd.DataFrame(
```
Comment (santhosh-thangavel) [inline csvqb/csvqb/tests/unit/utils/qb/test_standardise.py:16]: Why do we convert QbCube’s DataFrame columns(appropriate columns) to Categorical data type?

Comment (robons) [reply csvqb/csvqb/tests/unit/utils/qb/test_standardise.py:16]: Because it allows us to do a more performant replacement of the column's values.

**Comment by santhosh-thangavel** [comment]
@robons Thanks

**Review by santhosh-thangavel — APPROVED** [review]

**Pull request merged**